specfile/utils.py

```diff
@@ -1,10 +1,55 @@
 # Copyright Contributors to the Packit project.
 # SPDX-License-Identifier: MIT
 
+import re
 import urllib.parse
+from dataclasses import dataclass
 from pathlib import Path
 
 
+@dataclass
+class EVR:
+    epoch: int = 0
+    version: str = ""
+    release: str = ""
+
+    def __str__(self) -> str:
+        epoch = f"{self.epoch}:" if self.epoch > 0 else ""
+        release = f"-{self.release}" if self.release else ""
+        return f"{epoch}{self.version}{release}"
+
+    @staticmethod
+    def from_string(evr: str) -> "EVR":
+        m = re.match(r"^(?:(\d+):)?(.*?)(?:-([^-]*))?$", evr)
+        if not m:
+            return EVR()
```

Comment:
Is an "empty version" semantically valid? I'd maybe prefer throwing an error here.

Reply:
Hm, I thought it was, but it isn't. Let me fix that.